File: apps/ios/Sources/MobileCompanion/CompanionContracts.swift
import Foundation

enum MobileReleaseCapability: String, CaseIterable, Codable {
    case approvalsInbox = "approvals_inbox"
    case pollingNotifications = "polling_notifications"
    case recentSessions = "recent_sessions"
    case safeUrlOpen = "safe_url_open"
    case voiceNote = "voice_note"
}

enum MobileInboxPriority: String, Codable {
    case critical
    case high
    case medium
    case low
}

struct MobileHandoffTarget: Codable, Equatable {
    var path: String
    var intent: String?
    var requiresFullConsole: Bool
}

struct MobileSessionRecap: Codable, Equatable {
    var title: String
    var preview: String?
    var lastSummary: String?
    var lastIntent: String?
    var lastRunState: String?
    var pendingApprovals: Int
    var handoffRecommended: Bool
}

struct MobileApprovalSummary: Codable, Equatable {
    var approvalId: String
    var title: String
    var summary: String
    var priority: MobileInboxPriority
    var handoffTarget: MobileHandoffTarget
}

struct MobileBootstrapContract: Codable, Equatable {
    var localeOptions: [String] = ["en", "qps-ploc"]
    var defaultLocale: String = "en"
    var releaseScope: Set<MobileReleaseCapability> = [
        .approvalsInbox,
        .pollingNotifications,
        .recentSessions,
        .safeUrlOpen,
        .voiceNote,
    ]
    var defaultPollIntervalMs: Int = 45_000
    var quietHoursSupported: Bool = true
    var revokeSupported: Bool = true
}

File: apps/android/app/src/main/kotlin/io/palyra/mobile/companion/CompanionContracts.kt
package io.palyra.mobile.companion

enum class MobileReleaseCapability {
    APPROVALS_INBOX,
    POLLING_NOTIFICATIONS,
    RECENT_SESSIONS,
    SAFE_URL_OPEN,
    VOICE_NOTE,
}

enum class MobileInboxPriority {
    CRITICAL,
    HIGH,
    MEDIUM,
    LOW,
}

data class MobileHandoffTarget(
    val path: String,
    val intent: String? = null,
    val requiresFullConsole: Boolean = false,
)

data class MobileSessionRecap(
    val title: String,
    val preview: String? = null,
    val lastSummary: String? = null,
    val lastIntent: String? = null,
    val lastRunState: String? = null,
    val pendingApprovals: Int = 0,
    val handoffRecommended: Boolean = false,
)

data class MobileApprovalSummary(
    val approvalId: String,
    val title: String,
    val summary: String,
    val priority: MobileInboxPriority,
    val handoffTarget: MobileHandoffTarget,
)

data class MobileBootstrapContract(
    val localeOptions: List<String> = listOf("en", "qps-ploc"),
    val defaultLocale: String = "en",
    val releaseScope: Set<MobileReleaseCapability> = setOf(
        MobileReleaseCapability.APPROVALS_INBOX,
        MobileReleaseCapability.POLLING_NOTIFICATIONS,
        MobileReleaseCapability.RECENT_SESSIONS,
        MobileReleaseCapability.SAFE_URL_OPEN,
        MobileReleaseCapability.VOICE_NOTE,
    ),
    val defaultPollIntervalMs: Long = 45_000,
    val quietHoursSupported: Boolean = true,
    val revokeSupported: Boolean = true,
)
